x-pack/legacy/plugins/monitoring/public/components/cluster/overview/apm_panel.js

```diff
@@ -39,35 +39,47 @@ export function ApmPanel(props) {
+  const setupModeAPMData = get(setupMode.data, 'apm');
   let setupModeInstancesData = null;
   if (setupMode.enabled && setupMode.data) {
-    const migratedNodesCount = Object.values(setupModeApmData.byUuid).filter(node => node.isFullyMigrated).length;
-    let totalNodesCount = Object.values(setupModeApmData.byUuid).length;
-    if (totalNodesCount === 0 && get(setupMode.data, 'apm.detected.mightExist', false)) {
-      totalNodesCount = 1;
-    }
+    const {
+      totalUniqueInstanceCount,
+      totalUniqueFullyMigratedCount,
+      totalUniquePartiallyMigratedCount
+    } = setupModeAPMData;
+    const hasInstances = totalUniqueInstanceCount > 0 || get(setupModeAPMData, 'detected.mightExist', false);
+    const allMonitoredByMetricbeat = totalUniqueInstanceCount > 0 &&
+      (totalUniqueFullyMigratedCount === totalUniqueInstanceCount || totalUniquePartiallyMigratedCount === totalUniqueInstanceCount);
+    const internalCollectionOn = totalUniquePartiallyMigratedCount > 0;
+    if (hasInstances && (!allMonitoredByMetricbeat || internalCollectionOn)) {
+      let tooltipText = null;
 
-    const badgeColor = migratedNodesCount === totalNodesCount
-      ? 'secondary'
-      : 'danger';
+      if (!allMonitoredByMetricbeat) {
+        tooltipText = i18n.translate('xpack.monitoring.cluster.overview.apmPanel.setupModeNodesTooltip.oneInternal', {
+          defaultMessage: `There's at least one server that isn't being monitored using Metricbeat. Click the flag
+          icon to visit the servers listing page and find out more information about the status of each server.`
+        });
+      }
+      else if (internalCollectionOn) {
+        tooltipText = i18n.translate('xpack.monitoring.cluster.overview.apmPanel.setupModeNodesTooltip.disableInternal', {
+          defaultMessage: `All servers are being monitored using Metricbeat but internal collection still needs to be turned
+          off. Click the flag icon to visit the servers listing page and disable internal collection.`
+        });
+      }
 
-    setupModeInstancesData = (
-      <EuiFlexItem grow={false}>
-        <EuiToolTip
-          position="top"
-          content={i18n.translate('xpack.monitoring.cluster.overview.apmPanel.setupModeNodesTooltip', {
-            defaultMessage: `These numbers indicate how many detected monitored APM servers versus how many ` +
-            `detected total APM servers. If there are more detected APM servers than monitored APM servers, click the Nodes ` +
-            `link and you will be guided in how to setup monitoring for the missing node.`
-          })}
-        >
-          <EuiBadge color={badgeColor}>
-            {migratedNodesCount}/{totalNodesCount}
-          </EuiBadge>
-        </EuiToolTip>
-      </EuiFlexItem>
-    );
+      setupModeInstancesData = (
+        <EuiFlexItem grow={false}>
+          <EuiToolTip
+            position="top"
+            content={tooltipText}
```

Comment:
What happens if none of the conditions above were met and the `tooltipText` is `null`. Should we maybe also set `setupModeInstancesData` to `null`? Same concern for Beats and Logstash panels bellow

Reply:
It shouldn't ever happen due to how the `if` is structured above, but I can remove the `else if` to make it more clear, if that helps?

Reply:
Yep, I see it now 👍